modules/oe_theme_helper/src/Plugin/Field/FieldFormatter/FeaturedMediaFormatter.php

```diff
@@ -165,12 +167,16 @@ protected function viewElement(FieldItemInterface $item, string $langcode): arra
 
     // Retrieve the correct media translation.
     $media = $this->entityRepository->getTranslationFromContext($media, $langcode);
-
-    // Caches are handled by the formatter usually. Since we are not rendering
-    // the original render arrays, we need to propagate our caches to the
-    // oe_theme_helper_featured_media template.
     $cacheability->addCacheableDependency($media);
 
+    // Run access checks on the media entity.
+    $access = $media->access('view', NULL, TRUE);
+    $cacheability->addCacheableDependency($access);
+    if (!$access->isAllowed()) {
+      $cacheability->applyTo($build);
+      return [];
```

Comment:
```suggestion
      return $build;
```
Otherwise cache metadata isn't added.